python/lsst/meas/extensions/piff/wcs_wrapper.py

```diff
@@ -0,0 +1,219 @@
+            origin = galsim._PositionD(0.0, 0.0)
+        self._pix_to_sky = pix_to_sky
+        self._origin = origin
+        self._color = None
+
+    @property
+    def origin(self):
+        """The image coordinate position to use as the origin.
+        """
+        return self._origin
+
+    def _radec(self, x, y, color=None):
+        x1 = np.atleast_1d(x)
+        y1 = np.atleast_1d(y)
+
+        ra, dec = self._pix_to_sky.pixelToSkyArray(x1, y1)
+
+        if np.ndim(x) == np.ndim(y) == 0:
+            return ra[0], dec[0]
+        else:
+            # Sanity checks that the inputs are the same shape.
+            assert np.ndim(x) == np.ndim(y)
+            assert x.shape == y.shape
+            return ra, dec
+
+    def _xy(self, ra, dec, color=None):
+        r1 = np.atleast_1d(ra)
+        d1 = np.atleast_1d(dec)
+
+        x, y = self._pix_to_sky.skyToPixelArray(r1, d1)
+
+        if np.ndim(ra) == np.ndim(dec) == 0:
+            return x[0], y[0]
+        else:
+            # Sanity checks that the inputs are the same shape.
+            assert np.ndim(ra) == np.ndim(dec)
+            assert ra.shape == dec.shape
+            return x, y
+
+    def _newOrigin(self, origin):
+        """Return a new CelestialWcsWrapper with new origin.
+
+        Parameters
+        ----------
+        origin : `galsim.PositionD`, optional
+            Origin in image coordinates.
+
+        Returns
+        -------
+        ret : `CelestialWcsWrapper`
+            Transformed WCS.
+        """
+        return CelestialWcsWrapper(self._pix_to_sky, origin=origin)
+
+
+class UVWcsWrapper(galsim.wcs.EuclideanWCS):
+    """Wrap a `lsst.afw.geom.TransformPoint2ToPoint2[2->2]` in a GalSim WCS.
+
+    Parameters
+    ----------
+    pix_to_field : `lsst.afw.geom.TransformPoint2ToPoint2[2->2]`
+        Transform to wrap.  Most likely PIXELS -> FIELD_ANGLE, but other 2D ->
+        2D transforms should be possible.
+    origin : `galsim.PositionD`, optional
+        Origin in image coordinates.
+    world_origin : `galsim.PositionD`, optional
+        Origin in world coordinates.
+    """
+    _rad_to_arcsec = 206264.80624709636
+
+    def __init__(self, pix_to_field, origin=None, world_origin=None):
+        # GalSim EuclideanWCS assumes
+        #     u = ufunc(x-x0, y-y0) + u0
+        #     v = vfunc(x-x0, y-y0) + v0
+        # where u,v are world (likely field angles), and (x, y) are pixels.
+        # GalSim also assumes that origin = x0, y0 and world_origin = u0, v0.
+        # I might have naively thought that uv(origin) == world_origin, but
+        # that appears to not be required. So we're just going to leave both
+        # free. We'll default initialize down below with both set to (0.0, 0.0)
+        # though, so ufunc is just mapping[0] and vfunc is mapping[1].
+
+        if origin is None:
+            origin = galsim._PositionD(0.0, 0.0)
+        if world_origin is None:
+            world_origin = galsim._PositionD(0.0, 0.0)
```

Comment:
Do these need to be `_PositionD` with a leading underscore? `galsim` being an external code to the codebase, we should avoid using "private" attributes.

Reply:
Yeah, I'm not sure.  The difference between `_PositionD` and `PositionD` here is speed.  `_PositionD` does have a [docstring](https://github.com/GalSim-developers/GalSim/blob/releases/2.4/galsim/position.py#L236), but doesn't show up in the [online documentation](https://galsim-developers.github.io/GalSim/_build/html/pos.html).  From my special point of view as a GalSim dev, I'm not at all worried about it's stability.  But I realize not everyone who make look at this will appreciate that.  If PSFs were less tight-loopy then I'd probably say do the switch, but given that they are, I think it's better to leave this (and similar elsewhere in this file) as is.  Would adding a comment here help maybe?

Reply:
Offline we decided a comment was sufficient here.